run-elrond-tests.py

```diff
@@ -73,47 +79,111 @@ def mandos_int_to_kint(mandos_int: str):
+        return bytes(raw_str[2:], 'ascii')
+
+    # address
+    if raw_str.startswith(addr_prefix):
+        padded_addr = raw_str[len(addr_prefix):].ljust(32, '_')
         padded_addr_bytes = bytes(padded_addr[:32], 'ascii')
         return padded_addr_bytes
-    if argument[0] == '+' or argument[0] == '-':
-        # encode signed integer
+
+    # fixed width number
+    if raw_str.startswith(u64_prefix):
+        return mandos_interpret_as_uint_fixedwidth(raw_str[len(u64_prefix):], 8)
+    if raw_str.startswith(u32_prefix):
+        return mandos_interpret_as_uint_fixedwidth(raw_str[len(u32_prefix):], 4)
+    if raw_str.startswith(u16_prefix):
+        return mandos_interpret_as_uint_fixedwidth(raw_str[len(u16_prefix):], 2)
+    if raw_str.startswith(u8_prefix):
+        return mandos_interpret_as_uint_fixedwidth(raw_str[len(u8_prefix):], 1)
+
+    # signed integer
+    if raw_str.startswith('+') or raw_str.startswith('-'):
         try:
-            int_num = int(argument.replace(',', ''))
-            return int_num.to_bytes(length=(8 + (int_num + (int_num < 0)).bit_length()) // 8, byteorder='big', signed=True)
+            num_int, num_len = convert_string_to_sint(raw_str)
+            return num_int.to_bytes(length=num_len, byteorder='big', signed=True)
         except ValueError:
             pass
+
+    # unsigned integer
     try:
-        # encode unsigned integer
-        int_num = int(argument.replace(',', ''))
-        return int_num.to_bytes((int_num.bit_length() + 7) // 8, 'big')
+        num_int, num_len = convert_string_to_uint(raw_str)
+        return num_int.to_bytes(num_len, 'big')
     except ValueError:
         pass
 
-    raise ValueError("Argument type not yet supported: %s" % argument)
+    raise ValueError("Argument type not yet supported: %s" % raw_str)
+
+def mandos_interpret_as_uint_fixedwidth(raw_str: str, width: int):
+    num_int, _ = convert_string_to_uint(raw_str)
+    return num_int.to_bytes(width, byteorder='big')
+
+def convert_string_to_uint(raw_str: str):
+    num_str = raw_str.replace('_', '')
+    num_str = num_str.replace(',', '')
+
+    if num_str.startswith('0x') or num_str.startswith('0X'):
+        num_str = num_str[2:]
+        num_len = len(num_str) // 2
+        if num_len == 0:
+            return (0, 0)
+        else:
+            num_int = int(num_str, 16)
+            return (num_int, num_len)
+
+    if num_str.startswith('0b') or num_str.startswith('0B'):
+        num_str = num_str[2:]
+        num_len = len(num_str) // 8
+        if num_len == 0:
+            return (0, 0)
+        else:
+            num_int = int(num_str, 2)
+            return (num_int, num_len)
+
+    num_int = int(num_str)
+    if (num_int < 0):
+        raise ValueError("Negative number not allowed in this context: %s" % raw_str)
+    num_len = (num_int.bit_length() + 7) // 8
+    return (num_int, num_len)
+
+def convert_string_to_sint(raw_str: str):
+    num_int, _ = convert_string_to_uint(raw_str[1:])
```

Comment:
This won't work if there is no initial `-` though? If it's `0xabcd` it will be truncated to `xabcd`

I think you want something like:
```
if raw_str.startswith('-'):
  sign = -1
  raw_str = raw_str[1:]
else:
  sign = 1
num_int, _ = sign * convert_string_to_uint(raw_str)
num_len = (8 + (num_int + (num_int < 0)).bit_length()) // 8
```